src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,578 @@
+    mediaSource.addEventListener('sourceopen', sourceOpen, { once: true });
+
+    function sourceOpen() {
+      URL.revokeObjectURL(video.src);
+
+      const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+      sourceBuffer.appendBuffer(data);
+
+      video.play().then(_ => {
+        // TODO: Fetch the rest of the video when user starts playing video.
+      });
+    }
+  });
+}
+```
+
+Warning: For cross-origin resources, make sure your CORS headers are set
+properly as opaque responses retrieved with `fetch(videoFileUrl, { mode:
+'no-cors' })` are not allowed to feed any video or audio element.
+
+### Create Range responses with a Service Worker
+
+Now what if you have fetched an entire video file and saved it previously in
+the Cache API. When browser sends a HTTP Range request, you certainly don't
+want to bring the entire video into renderer memory as the Cache API doesn't
+support Range responses yet.
+
+So let me show how to intercept these requests and return a customized Range
+response from a service worker.
+
+```
+addEventListener('fetch', event => {
+  event.respondWith(loadFromCacheOrFetch(event.request));
+});
+
+function loadFromCacheOrFetch(request) {
+  // Search through all available caches for this request.
+  return caches.match(request)
+  .then(response => {
+
+    // Fetch from network if it's not already in the cache.
+    if (!response) {
+      return fetch(request);
+      // Note that we may want to add the response to the cache and return
+      // network response in parallel as well.
+    }
+
+    // Browser sends a HTTP Range request. Let's provide one reconstructed
+    // manually from the cache.
+    if (request.headers.has('range')) {
+      return response.blob()
+      .then(data => {
+
+        // Get start position from Range request header.
+        const pos = Number(/^bytes\=(\d+)\-$/g.exec(request.headers.get('range'))[1]);
+        const options = {
+          status: 206,
+          statusText: 'Partial Content',
+          headers: response.headers
+        }
+        const slicedResponse = new Response(data.slice(pos), options);
+        slicedResponse.setHeaders('Content-Range': 'bytes ' + pos + '-' +
+            (data.size - 1) + '/' + data.size);
+        slicedResponse.setHeaders('X-From-Cache': 'true');
+
+        return slicedResponse;
+      });
+    }
+
+    return response;
+  }
+}
+```
+
+It is important to note that I used `response.blob()` to recreate this sliced
+response as this simply gives me a handle to the file while
+`response.arrayBuffer()` brings the entire file into renderer memory.
+
+My custom `X-From-Cache` HTTP header can be used to know whether this request
+came from the cache or from the network. It can be used by a player such as
+[ShakaPlayer] to ignore the response time as an indicator of network speed.
+
+<div class="video-wrapper">
+  <iframe class="devsite-embedded-youtube-video" data-video-id="f8EGZa32Mts"
```

Comment:
Reminder: @johnpallett mentioned to make sure the video is updated (NOT YouTube) (https://github.com/google/WebFundamentals/pull/4744#issuecomment-315051645)